plugins/logs/plugin.go

```diff
@@ -569,11 +569,18 @@ func (p *Plugin) Reconfigure(_ context.Context, config interface{}) {
 	<-done
 }
 
-// Trigger returns the plugin's trigger channel that can be used to control
-// when the plugin attempts to upload a new decision log in manual triggering
-// mode.
-func (p *Plugin) Trigger() chan chan struct{} {
-	return p.trigger
+// Trigger can be used to control when the plugin attempts to upload
+// a new decision log in manual triggering mode.
+func (p *Plugin) Trigger(ctx context.Context) {
+	done := make(chan struct{})
+	p.trigger <- done
+
+	select {
+	case <-done:
+		return
+	case <-ctx.Done():
```

Comment:
Same thing here, would be nice to get the context propagated to the client shipping the logs.

Reply:
Done